snuba/cli/subscriptions.py

```diff
@@ -43,6 +48,9 @@
     type=click.Choice(DATASET_NAMES),
     help="The dataset to target",
 )
+@click.option(
+    "--entity", "entity_name", help="The entity to target",
+)
```

Comment:
I think at least the executor should be simple enough that a shared executor pool for multiple entities in a dataset should be reasonably straightforward

Reply:
@fpacifici 

>Does this mean we will have to run one subscriptions consumer cluster per metrics entity ?

I thought that was already the case given that the default entity was always extracted as per https://github.com/getsentry/snuba/pull/2243/files#diff-11204424e2e0a0540fa9eadb5dbaf07391016ab4d94ad0818abdfb2e9a5bfaebL132
I just gave the choice to use a specific entity

@lynnagara 
> since metrics never even writes a commit log now

There is a PR open for that https://github.com/getsentry/snuba/pull/2222/files#diff-93e400870e85f66f3db6afceb3d7ea01c4cf66788c389e1c5a50eca0eb0b78c9R78-R84

Reply:
Moving this to https://github.com/getsentry/snuba/pull/2258

Reply:
> What I don't understand is how can all metrics share the same steam loader when they have different processors?

What matters for subscriptions consumer is whether they share the same commit log, which they do as they are in the same multi storage consumer. Two consumers that share the same topic and commit log can, in principle, share the subscriptions consumer. The problem is the subscriptions storage which is partitioned by entity, but that is an implementation issue not a conceptual issue.

> Why don't we want to allow one subscriptions consumer cluster per metrics entity?
Because it means allocating, maintaining and troubleshooting  4 clusters instead of one, while 1 will be able to deal with this for a long time. 

> "load multiple storages for multiple entities in the same dataset" Not sure I understand what this means
The subscriptions storage is partitioned by entity. The subscription consumer can only work with one at a time now. This is an implementation detail and not a conceptual issue. So we would have to make that change to run on one cluster. 

Now @lynnagara made the change to allow this to work in the new architecture (multiple schedulers but one executor cluster). So the situation will be better there. 
Do you need alerts on all types of metrics as of now? We can deploy only the cluster for the entities we need if you are not supporting alerts on all metrics types yet.

Reply:
I made a fairly small change `SubscriptionWithTick` so that it also stores the entity name (as that will no longer be clear when multiple entities are being passed in via the CLI. You may need to rebase this PR to get those tests and mypy passing here.

Reply:
@lynnagara 
Sounds good thanks.
After reading this https://github.com/getsentry/snuba/pull/2255, I am assuming this is no longer required -> https://github.com/getsentry/snuba/pull/2222/ because its the old infra, and instead I should put up a new PR where the subscription scheduler supports the new metrics entities?

@fpacifici 
Thanks for clarification.

> Do you need alerts on all types of metrics as of now? We can deploy only the cluster for the entities we need if you are not supporting alerts on all metrics types yet.

Yes that's fine for now. For Crash Rate alerts, we only need to support `metrics_counters` (for sessions crash free rate) and `metrics_sets` (for users crash free rate). That being said, like I mentioned in our slack thread I am meeting with product early next week and I will kick start a tech spec for future metric alerts to share with you then